internal/ingress/controller/template/template.go

```diff
@@ -86,6 +92,77 @@ func NewTemplate(file string) (*Template, error) {
 	}, nil
 }
 
+func cleanConf(in *bytes.Buffer, out *bytes.Buffer) {
+	depth := 0
+	lineStarted := false
+	emptyLineWritten := false
+	state := stateCode
+	for {
+		c, err := in.ReadByte()
+		if err != nil {
+			if err == io.EOF {
+				return
+			}
+			panic(err)
```

Comment:
I got it, but still I don't like the idea of something internally panic'ing. If this behavior changes in a future, we might end up with this panic doing a world stop into ingress nginx :)

It would be better to return this error and check if this is not nil, even if we know that now it will always be nil, if this changes in a future Go compilation we won't get some strange magic error :)